src/core/Akka/Actor/Props.cs

```diff
@@ -466,6 +466,25 @@ public Props WithRouter(RouterConfig routerConfig)
             copy.Deploy = Deploy.WithRouterConfig(routerConfig);
             return copy;
         }
+        
+        /// <summary>
+        ///     Creates a new <see cref="Akka.Actor.Props" /> with a given stash size.
+        ///     <note>
+        ///         This method is immutable and returns a new instance of <see cref="Akka.Actor.Props" />.
+        ///     </note>
+        /// </summary>
+        /// <remarks>
+        /// If the actor doesn't use an <see cref="IStash"/> or if it uses an <see cref="IWithUnboundedStash"/>, then this
+        /// setting will be ignored.
+        /// </remarks>
+        /// <param name="stashCapacity">The stash size to use when creating the actor.</param>
+        /// <returns>A new <see cref="Akka.Actor.Props" /> with the provided stash size..</returns>
+        public Props WithStashCapacity(int stashCapacity)
```

Comment:
Add overload for copying in `StashCapacity` without having to explicitly allocate a new `Deploy` each time (consistent with other settings.)